core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -641,17 +710,17 @@ private long remainingInternal() {
    */
   private void seekInternalWithCachingPartiallyReadBlock(long pos) throws IOException {
     // Precompute this because mPos will be updated several times in this function.
-    final boolean isInCurrentBlock = pos / mBlockSize == mPos / mBlockSize;
+    final boolean isInCurrentBlock = isInCurrentBlock(pos);
 
     if (isInCurrentBlock) {
       // the read is within the current block update stream to refresh the streams
-      updateStreams();
+      updateStreamsOnRead();
       if (isReadFromLocalWorker()) {
         // no need to partial cache the current block, and the seek is within the block
         // so directly seeks to position.
         mPos = pos;
-        // updateStreams is necessary when pos = mFileLength.
-        updateStreams();
+        // updateStreamsOnRead is necessary when pos = mFileLength.
```

Comment:
This comment says `updateStreamsOnRead` but the line below says `updateStreamsOnSeek`. Is that intentional?

Reply:
updated